.github/workflows/unit-tests.yml

```
@@ -12,46 +12,25 @@ jobs:
       fail-fast: false
       matrix:
         python-version:
-          - '3.10'
-          - '3.11'
-          - '3.12'
-          - '3.13'
-        poetry-version: ["1.8.3"]
+          - "3.10"
+          - "3.11"
+          - "3.12"
+          - "3.13"
     steps:
-    - uses: actions/checkout@11bd71901bbe5b1630ceea73d27597364c9af683 # v4.2.2
-      with:
-        fetch-depth: 0 # need all versions to be able to access server tree commit logs
-        persist-credentials: false
-    - uses: actions/setup-python@0b93645e9fea7318ecaed2b359559ac225c90a2b # v5.3.0
-      with:
-        python-version: ${{ matrix.python-version }}
-    - uses: awalsh128/cache-apt-pkgs-action@5902b33ae29014e6ca012c5d8025d4346556bd40 # latest
-      with:
-        packages: python3-dev libldap2-dev libsasl2-dev
-        version: 1.0
-    - name: Load cached Poetry Binary
-      id: cached-poetry-binary
-      uses: actions/cache@1bd1e32a3bdc45362d1e726936510720a7c30a57 # v4.2.0
-      with:
-        path: ~/.local
-        key: poetry-${{ runner.os }}-${{ matrix.python-version }}-${{ matrix.poetry-version }}
-    - name: Install Poetry
-      if: steps.cached-poetry-binary.outputs.cache-hit != 'true'
-      uses: snok/install-poetry@76e04a911780d5b312d89783f7b1cd627778900a # v1.4.1
-      with:
-        version: ${{ matrix.poetry-version }}
-        virtualenvs-create: true
-        virtualenvs-in-project: true
-    - name: Load cached venv
-      id: cached-poetry-dependencies
-      uses: actions/cache@1bd1e32a3bdc45362d1e726936510720a7c30a57 # v4.2.0
-      with:
-        path: .venv
-        key: venv-${{ runner.os }}-${{ matrix.python-version }}-${{ hashFiles('**/pyproject.toml') }}
-    - name: Install dependencies
-      if: steps.cached-poetry-dependencies.outputs.cache-hit != 'true'
-      shell: bash
-      run: poetry install --no-interaction --no-root ${{ inputs.install-args }}
-    - name: Test with pytest
-      run: |
-        poetry run pytest -rA
+      - uses: actions/checkout@v5
```

the latest version is v6, we probably can try it

```suggestion
      - uses: actions/checkout@v6
```